internal/controller/object/object.go

```diff
@@ -246,6 +243,12 @@ func (c *external) Create(ctx context.Context, mg resource.Managed) (managed.Ext
 	}
 
 	c.logger.Debug("Creating", "resource", cr)
+
+	if policy := cr.Spec.ManagementPolicy; policy == v1alpha1.ObserveDelete || policy == v1alpha1.Observe {
```

Comment:
Having a simple function to decide whether an action is allowed or not would improve readability, something like: (prefer defining an enum instead of string as input though)

```
if !isAllowedTo("Create") {

}
```